### PR TITLE
Adding Garbage Collection for NMC objects for deleted Nodes

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
This implemented by setting Node as the owner of the NMC. In case Node is deleted, NMC will also be deleted

### CODE REVIEW

File: internal/controllers/module_nmc_reconciler.go [hunk 1]
```diff
@@ -236,17 +236,17 @@ func (mnrh *moduleNMCReconcilerHelper) enableModuleOnNode(ctx context.Context, m
 	}
 
 	nmc := &kmmv1beta1.NodeModulesConfig{
-		ObjectMeta: metav1.ObjectMeta{Name: nodeName},
+		ObjectMeta: metav1.ObjectMeta{Name: node.Name},
 	}
 
 	opRes, err := controllerutil.CreateOrPatch(ctx, mnrh.client, nmc, func() error {
-		return mnrh.nmcHelper.SetModuleConfig(nmc, mld, &moduleConfig)
+		return mnrh.nmcHelper.SetModuleConfig(nmc, mld, &moduleConfig, node)
```
Comment (qbarrand) [inline internal/controllers/module_nmc_reconciler.go:243]: I think I would rather call `controllerutil.SetOwnerReference` here. It does not belong to `SetModuleConfig` in my opinion, which should just manipulate the `spec.modules` list.

Comment (yevgeny-shnaidman) [reply internal/controllers/module_nmc_reconciler.go]: Fixed

**Pull request merged**